File: string_processing/pl/string_splits_01.pl
#!/usr/bin/perl

#######################################################
## Splitting strings in PL
## author: vladimir kulyukin
#######################################################

use strict;
use warnings;

my $text_01 = 'word1 word2 word3';
my $text_02 = 'word1, word2; word3';

sub split_test_01 { print "$_\n" foreach(split(/ /, $_[0])); }

split_test_01($text_01);
split_test_01($text_02);		    

sub split_test_02 { print "$_\n" foreach(split(/;/, $_[0])); }

split_test_02($text_01);
split_test_02($text_02);

sub split_test_n {
    my ($str, $n) = @_;
    print "$_\n" foreach(split(/ /, $str, $n));
}

# tests;
#print "============\n";
#split_test_n($text_01, 1);
#split_test_n($text_01, 2);
#split_test_n($text_01, 3);

File: string_processing/py/string_splits_01.py
#!/usr/bin/python

#######################################################
## Splitting strings in Py
## author: vladimir kulyukin
#######################################################

text_01 = 'word1 word2 word3'
text_02 = 'word1, word2; word3'

def split_test_01(txt):
    for s in txt.split(' '):
        print s

split_test_01(text_01)
split_test_01(text_02)

def split_test_02(txt):
    for s in txt.split(';'):
        print s

split_test_02(text_01)
split_test_02(text_02)
